File: Crawler_LogFolder/crawler_log.go
package crawlerlogfolder

// var mapCd = map[string]int{"../": -1, "./": 0}

func minOperations(logs []string) int {
	// minReturn := 0
	//
	// for _, operation := range logs {
	// 	val, exist := mapCd[operation]
	//
	// 	if !exist {
	// 		minReturn++
	// 	} else {
	// 		minReturn += val
	// 		if minReturn < 0 {
	// 			minReturn = 0
	// 		}
	// 	}
	// }
	//
	// return minReturn

	stack := []byte{}

	for _, operation := range logs {
		if operation == "../" {
			if len(stack) > 0 {
				stack = stack[1:]
			}
		} else if operation == "./" {
			continue
		} else {
			stack = append(stack, 1)
		}
	}

	return len(stack)
}

File: Crawler_LogFolder/crawler_log.py
from typing import List


class Solution:
    def minOperations(self, logs: List[str]) -> int:
        stack: List[bool] = []
        index = 0

        while index < len(logs):

            if logs[index] == "../":
                if len(stack) > 0:
                    stack = stack[1:]
            elif logs[index] == "./":
                pass
            else:
                stack.append(True)

            index += 1

        return len(stack)
